certificate-manager/src/main/java/io/strimzi/certs/OpenSslCertManager.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An OpenSSL based certificates manager
+ */
+public class OpenSslCertManager implements CertManager {
+
+    private static final Logger log = LogManager.getLogger(OpenSslCertManager.class);
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, int days) throws IOException {
+        generateSelfSignedCert(keyFile, certFile, null, days);
+    }
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, Subject sbj, int days) throws IOException {
+
+        List<String> cmd = new ArrayList<>(Arrays.asList("openssl", "req", "-x509", "-new", "-days", String.valueOf(days), "-batch", "-nodes",
+                "-out", certFile.getAbsolutePath(), "-keyout", keyFile.getAbsolutePath()));
+
+        File sna = null;
+        if (sbj != null) {
+
+            if (sbj.subjectAltNames() != null && sbj.subjectAltNames().size() > 0) {
+
+                // subject alt names need to be in an openssl configuration file
+                File file = new File(getClass().getClassLoader().getResource("openssl.conf").getFile());
```

Comment:
It works at least in the unit tests because we don't have a real use case for now but in any case I changed to load data as resource stream. In any case I put them into a `File` used by the methods exposed in the cert manager class.